### ISSUE TITLE
remove non-free dependency com.google.android.play.core.splitcompat.SplitCompatApplication

### ISSUE THREAD

**Issue by TheOneWithTheBraid** [flutter/flutter#104219]
## Use case

The implementation of `com.google.android.play.core.splitcompat.SplitCompatApplication` in https://github.com/flutter/engine/blob/main/shell/platform/android/io/flutter/embedding/android/FlutterPlayStoreSplitApplication.java and https://github.com/flutter/engine/blob/main/shell/platform/android/io/flutter/app/FlutterPlayStoreSplitApplication.java should be made optional and not included by default in the android builds.

Those implementations should be optionally imported into built APKs / appbundles in order to shrink the built applications size. Moreover, making those implementations would result in apps with entirely open source dependencies; which might be important for high-security certifications.

## Proposal

The stated implementations should only be included in the built applications in case deferred libraries are used. Otherwise, they should be left out in order to produce smaller builds.

**Comment by chinmaygarde** [comment]
cc @blasten

**Comment by dnfield** [comment]
For what it's worth, the final APK you produce will almost certainly not include this class if you are not referencing it and are using R8, which by default Flutter apps do.

Do you have evidence otherwise?

**Comment by TheOneWithTheBraid** [comment]
No, in case R8 is used, it isn't included. It's more a kind of philosophical question whether to include the GSF by default into all builds.
